docs/user_docs/concepts/intent

```diff
@@ -0,0 +1,90 @@
+# Intent
+
+Intent represents the operational intentions you want to deliver with Kusion. The intentions here is expected to contain all components during the SDLC, such as resources, dependencies and policies. All AppConfigurations written by application developers will also be converted into the Intent by Kusion module generators before Kusion acutally perform operations.
+
+## Purpose
+
+### Single Source of Truth
+
+In Kusion's workflow, platform engineer build Kusion modules and provide environment configurations, application developers choose Kusion modules they need and deploy operational intentions to an environment with related environment configurations. They can also input dynamic paramters like the container image when execute the `kusion build` command. So the final operational intentions includes configuratins written by application developers, environment configurations and dynamic inputs. Due to this reason, we introduce **Intent** to represent the SSoT(Single Source of Truth) of Kusion. It is the result of `kusion build` which contains all operational intentions from different sources.
+
+### Consistency
+
+By utilizing the Intent, all configurations and changes are stored in a single file. As the Intent is the input of Kusion, it ensures consistency across different environments whenever you execute Kusion with the same Intent file.
+
+### Rollback and Disaster Recovery
+
+We recommend to sotre the Intent in a version control systems like Git. This enables better change management practices and makes it simpler to roll back to previous versions if needed. In case of a failure or outage, having a validated Intent simplifies the rollback process, ensuring that the system can be quickly recovered.
```

Comment:
> Sources of Truth are the starting point of the pipeline. Whenever a SoT changes, we rerun the full pipeline, which creates a new incarnation. This process occurs continuously. Because some partitions undergo many changes, we coalesce regeneration. Therefore, a new incarnation might be caused by multiple SoT changes.
I would suggest deep dive into https://sre.google/workbook/configuration-design/, and think about this from the perspective of configuration data.